pithos/pithos.py

```diff
@@ -843,7 +845,18 @@ def update_song_row(self, song = None):
         if song:
             self.songs_model[song.index][1] = self.song_text(song)
             self.songs_model[song.index][2] = self.song_icon(song) or ""
-        return self.playing
+        if self.playing:
+            return True
+
+    def create_ui_loop(self):
+        if self.ui_loop_timer_id is not None:
```

Comment:
This likely should not compare against None. If it succeeds it will always return > 0 and if it fails it returns 0, in which case you *want* a new timeout.